File: PyNet/ga2230Net/SenderSocket.py
import socket

from Packet import Packet


class SenderSocket:
    """A socket handler for sending data.
    Implements a client that sends data to listeners.
    """
    def __init__(self, listener_ip: str, listener_port: int) -> None:
        """Creates a new sender socket

        :param listener_ip: the ip of the listener
        :type listener_ip: str
        :param listener_port: the port of the listener
        :type listener_port: int
        """
        self.sock: socket.socket = socket.socket(
            socket.AF_INET, socket.SOCK_DGRAM
        )
        self.listener_address = (listener_ip, listener_port)

    def send_packet(self, packet: Packet) -> None:
        """Sends a packet to a the listener.

        :param data: the packet
        :type data: Packet
         listener
        """
        self.sock.sendto(packet.serialize(), self.listener_address)

File: RIONet/ga2230net/src/main/java/org/ga2230net/SenderSocket.java
package org.ga2230net;

import java.net.DatagramSocket;
import java.net.DatagramPacket;
import java.net.InetAddress;
import java.io.IOException;

public class SenderSocket {
    private final DatagramSocket datagramSocket;

    private final int port;
    private final InetAddress listenerAddress;

    public SenderSocket(String listener_ip, int listener_port, int sender_port) throws IOException {
        this.port = listener_port;
        this.listenerAddress = InetAddress.getByName(listener_ip);
        datagramSocket = new DatagramSocket(sender_port);
    }

    /**
     * sends a packet to the listener
     *
     * @param packet the packet to send
     * @throws IOException if there was a problem while sending the packet
     */
    public void sendPacket(Packet packet) throws IOException {
        byte[] ser = packet.serialize();
        DatagramPacket pack = new DatagramPacket(
            ser, ser.length,
            listenerAddress, port
        );
        datagramSocket.send(pack);
    }
}